### PR TITLE
Link Control: persist advanced settings toggle state to preferences if available

### LINKED ISSUE #52216: The new design of the link UI in 16.1.0 has turned one click into four
## What problem does this address?
In 16.1.0 you "cleaned up" the UI for adding a link into a post. But I ALWAYS want to make the link click open in a new window, and now if I add a link I have to click back into the link, then click on "Advanced options" and then click on the open in a new window option.  I'd love for this to be cleaner and simpler, but can you please add some options into the preferences to allow these options on the link add to be defaulted to my preferences? 

## What is your proposed solution?
Add some global preferences for how the "add link" dialogue options should be defaulted.

### PR DESCRIPTION
**Pull request by scruffian**
This is an alternative to https://github.com/WordPress/gutenberg/pull/52321. Fixes #52216.

## What?
Once a user opens the advanced section of link control it will stay open until they close it again.

## Why?
This retains the UI state which will be useful for people who often use the "open in new tab" option.

## How?
Adds a new preference that gets toggled when you open the advanced section of link control.

## Testing Instructions
1. Add a paragraph
2. Add a link to some of the text
3. Open the advanced section of link control
4. Close the link editing UI
5. Reopen the link editing UI and confirm that the advanced section is open
6. Close the advanced option
7. Close the link editing UI
8. Reopen the link editing UI and confirm that the advanced section is still closed.
9. Also try reloading the edftor and checking the setting persists across reloads.
10. Also try in different editor environments and check the setting persists.

### Automated Tests

e2e tests  - `npm run test:e2e:playwright test/e2e/specs/editor/blocks/links.spec.js -- -g "toggle state of advanced link settings is preserved across editing links"`

Co-authored-by: Dave Smith <444434+getdave@users.noreply.github.com>

### CODE REVIEW

**Comment by draganescu** [comment]
I like the idea of this automagic behavior. In this PR the option does reset if I click a link that is already linked.

File: packages/block-editor/src/components/link-control/index.js
Comment (getdave) [inline packages/block-editor/src/components/link-control/index.js]: I don't think you should have any reference to the various editor implementations. This is the block editor package.

Rather you can pass in a "setter" function via settings which this component can access and use.

This is similar to how we access the preference


```
select( blockEditorStore ).getSettings()
				.linkControlAdvancedSettingsPreference,

```

File: packages/block-editor/src/components/link-control/index.js
Comment (getdave) [inline packages/block-editor/src/components/link-control/index.js]: This won't work as we expect. The preference should _only_ be set when the _user_ toggles the control. Otherwise the behaviour will be unpredictable.

File: packages/block-editor/src/components/link-control/index.js
Comment (getdave) [inline packages/block-editor/src/components/link-control/index.js]: I updated so the setter is provided by the block editor settings. This means any editor can provide their own preference store rather than having these hard coded to WP only as was previously the case.

File: packages/block-editor/src/components/link-control/index.js
Comment (getdave) [inline packages/block-editor/src/components/link-control/index.js:141]: We need the local state as a fallback in case that the preference isn't supplied. I considered using a local state as the value passed as the setting but I realised that is also no good because the editor might not pass the settings _at all_ so we cannot rely on their presence.

File: packages/block-editor/src/components/link-control/index.js
Comment (getdave) [reply packages/block-editor/src/components/link-control/index.js]: Can we determine which scenarios we want to call

- set local state
- set preference

I'm starting to consider that what we actually need is a `const` computed var for `shouldShowAdvanceSettings` which would need to be true for the drawer to show _at all_.

That way we ensure the drawer doesn't show up when it's not supposed to (as it does currently!).

**Comment by getdave** [comment]
> When you open the advanced section the Link UI state is reset - I don't know why but this is the wrong behaviour!

@scruffian please can you explain this a little more? Replication steps and exactly what you saw and expected? If you pull the latest changes does it still happen? 🙏 🙏

**Comment by getdave** [comment]
Needs unit test coverage before merging to test that the toggle behaviour works when a preference is available.

**Comment by getdave** [comment]
> > When you open the advanced section the Link UI state is reset - I don't know why but this is the wrong behaviour!
> 
> @scruffian please can you explain this a little more? Replication steps and exactly what you saw and expected? If you pull the latest changes does it still happen? 🙏 🙏

@scruffian @draganescu I think the reason this is happening is because toggling the setting resets the internal state which then causes the value syncing to run. We've been looking at this value syncing and whether it should be removed in [this PR](https://github.com/WordPress/gutenberg/pull/51387).

**Comment by scruffian** [comment]
> @scruffian please can you explain this a little more? Replication steps and exactly what you saw and expected? If you pull the latest changes does it still happen? 🙏 🙏

https://github.com/WordPress/gutenberg/assets/275961/15414788-d935-4184-a2ce-ddc7963ee837

**Comment by getdave** [comment]
I'm concerned about this change. I think adding this preference to the editor settings means that each time you toggle the "drawer" it will cause the memoization of the settings to be invalidated which will cause the editor to re-render. That's probably what's contributing to the state reset we're seeing.

I'm wondering if there are other options we could consider. For example, only _persisting_ the preference when the current `<LinkControl>` is "unmounted". Perhaps in an effect.

@Mamaduka I know you're really great an optimisations such as these. I'd greatly value your perspective here 🙇

**Comment by getdave** [comment]
With the change in https://github.com/WordPress/gutenberg/pull/52799/commits/facd271496b7484306ff892715518f26d6415169 it's harder to replicate the "wiping out state" bug, but here is a new video with replicate steps:


https://github.com/WordPress/gutenberg/assets/444434/2388c3f2-56f5-4d9d-98bf-e328580b07e5

File: packages/block-editor/src/components/link-control/index.js
Comment (getdave) [reply packages/block-editor/src/components/link-control/index.js]: I updated this so that we have a computed var.

File: packages/block-editor/src/components/link-control/index.js
Comment (getdave) [reply packages/block-editor/src/components/link-control/index.js]: ✅ I've fixed this.

**Comment by Mamaduka** [comment]
I think this is one of those options that doesn't have to be editor specific. Maybe we could use a general preference store and avoid funneling value/setter via settings.

The `Hint` components use `core` as the store name. Examples:

* https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/inspector-controls-tabs/settings-tab-hint.js
* https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/inserter/reusable-block-rename-hint.js

**Comment by getdave** [comment]
Thank you for highlighting this. 

The approach is interesting to me as that seems like coupling block editor code to a concept of "Core". What is Core in Drupal? It seems like WP specific code which I understood should not exist in the block editor package 🤔 

> avoid funneling value/setter via settings.

This would be preferable though...

**Comment by Mamaduka** [comment]
The preference scope can also be a `core/block-editor` if we want to use a package as scope name conventions. The preference package isn't WP-specific; you can use different persistence layers.

cc @talldan

**Comment by draganescu** [comment]
Can't help to wonder if this is the case where we actually need a simpler solution like for this component only just use local storage. What's wrong with that?

**Comment by getdave** [comment]
> Can't help to wonder if this is the case where we actually need a simpler solution like for this component only just use local storage. 

The preference layer can be configured to do that. But I assume you're talking about a standalone localstorage value for this UI only. 

> What's wrong with that?

I'm not sure that's a great precedent to set. We'd end up with a tonne of "one off" things set in local storage.

I think @Mamaduka is probably on to the right track with having a generic scope. But what I want to know is that another consumer of the `block-editor` package can somehow pass their own preferences system into the editor and have it work. If that's the case then we don't need to pass stuff via settings anymore and can simply interact with the registered preferences system.

**Comment by Mamaduka** [comment]
> But what I want to know is that another consumer of the block-editor package can somehow pass their own preferences system into the editor and have it work.

They can.  This is how WP does it - https://github.com/WordPress/wordpress-develop/blob/12a0205829c1411f16b4e216a713a153bcc62e2f/src/wp-includes/script-loader.php#L372-L388.

**Comment by getdave** [comment]
Ok that's brilliant. Let's try using `core/block-editor` as the scope here and removing the abstraction via settings. I'll try that shortly.

**Comment by getdave** [comment]
@scruffian @Mamaduka I've now updated to use the `core/block-editor` scope. The e2e test confirms the functionality but you should also test in browser.

@richtabor cc'ing you here as you've been working in/around the Link UI recently.

**Comment by draganescu** [comment]
> I'm not sure that's a great precedent to set. We'd end up with a tonne of "one off" things set in local storage.

I don't think this is true :) because there is code review and not everything is a precedent. For example there is the situation with the block inspector having settings and design tabs. Designing with it I would love to have the open tab persist when block selection changes. Should we another preference for that too?

I don't think the path here is _bad_ it just feels like we're skipping over simple solutions when they're quite fit for the problem - simple UI intelligent behavior data can be saved in localstorage and it's fine.

**Comment by richtabor** [comment]
It works as expected; the "Advanced" toggle is open if I leave it open when saving; and vise-versa.

**Comment by richtabor** [comment]
I would prefer if the Advanced button wasn't the admin color when opened though.

**Comment by getdave** [comment]
> I don't think the path here is _bad_ it just feels like we're skipping over simple solutions when they're quite fit for the problem - simple UI intelligent behavior data can be saved in localstorage and it's fine.

Thanks for the alternative perspective. I don't know how to draw a conclusion here because everything discussed is relatively subjective and thus the route we take is not clear cut. One way to break this deadlock is to try drawing on prior art within Gutenberg. Are there any "preferences" similar to this that use localstorage?

The only other thing in favour of utilising the `@wordpress/preferences` package is that in WordPress the setting will be persisted _across devices_ and for the current user _only_ (this is because it's saved to user meta). I can imagine frequent users of `Opens in new tab` (and future friends) won't want the setting changing on each new device. Probably not the end of the world but additional context to consider perhaps.

How can we unblock this decision?

**Comment by ricjcs** [comment]
Wouldn't it be possible, when clicking on the link, to open directly in the box with the options? That is, not having to click the "Edit" button.

With this solution, in my view, the problem practically remains. Only 1 click is avoided. It still takes a series of clicks to set a link to open in a new tab. And with a simple click it is not possible to see if it is set to open in a new tab or not.

Comparing with WP 6.2...

[Wordpress new tab.webm](https://github.com/WordPress/gutenberg/assets/11923491/39c7b74c-92b2-4c4b-a884-482e519b4c14)

WP 6.2 continues to be much more efficient.

**Comment by richtabor** [comment]
> Wouldn't it be possible, when clicking on the link, to open directly in the box with the options? That is, not having to click the "Edit" button. 

I do want to explore removing that interstitial step here: https://github.com/WordPress/gutenberg/issues/50892

**Comment by draganescu** [comment]
> How can we unblock this decision?

I am not blocking, as I said it's not bad to use the pref package either (it does have advantages). If it only gets every too complictated then we still have the simple solution.

**Comment by draganescu** [comment]
@getdave can this same technique here be applied for #51104?

**Comment by scruffian** [comment]
This looks good to me. I can't approve it, as it's my PR!

File: packages/block-editor/src/components/link-control/README.md
Comment (scruffian) [inline packages/block-editor/src/components/link-control/README.md:26]:
```suggestion
Consumers who wish to take advantage of this functionality should ensure that their block editor environment utilizes the [`@wordpress/preferences`](packages/preferences/README.md) package. By default the `<LinkControl>` component will attempt to persist the state of UI to a setting named `linkControlSettingsDrawer` with a scope of `core/block-editor`. If the preferences package is not available then local state is used and the setting will not be persisted.
```

**Comment by MaggieCabrera** [comment]
Not a blocker, but when the Advanced toggle is focused the border looks strange in this context:

Screenshot 2023-08-01 at 13 52 27

**Comment by MaggieCabrera** [comment]
> I would prefer if the Advanced button wasn't the admin color when opened though.

I think we still need to make this change

**Comment by scruffian** [comment]
> I think we still need to make this change

It's tracked in https://github.com/WordPress/gutenberg/issues/53069 as it's not related to this change.

**Comment by scruffian** [comment]
@kevin940726 we're stuck on why this test is failing. Do you have any insights?

**Comment by Mamaduka** [comment]
I would recommend checking the failure artifacts in Playwright trace viewer - https://trace.playwright.dev. From what I can tell, the test fails to select the "Gutenberg" text and edits the "WordPress".

Artifacts: https://github.com/WordPress/gutenberg/actions/runs/5726539229?pr=52799

## Screencast
[video.webm](https://github.com/WordPress/gutenberg/assets/240569/5c874e75-7f61-425e-9bb2-b4a62ee1ab65)

**Review by MaggieCabrera — APPROVED** [review]

**Comment by scruffian** [comment]
I think it would be good to get this into 6.3 if it's not too late...

**Comment by Mamaduka** [comment]
> I think it would be good to get this into 6.3 if it's not too late...

I think it's late to backport features in 6.3; the last RC was released yesterday.

**Comment by richtabor** [comment]
Great job on this folks. 🚀

**Comment by EW0JY** [comment]
Hi guys,

I appreciate the work you put into trying to fix the problem regarding links which were introduced with the latest 6.3 release.

I've dug into this the last few hours and saw that there are multiple threads dealing with possible solutions.

My question is this:

I'm a bit confused as to why this current thread here is closed and you guys consider it solved:

> Great job on this folks.

I'm afraid that it hasn't been resolved, here's why:

You still have to go through the following tedious process:

1. select text and "insert link" (crtl+k)
2. hit enter to create / insert the link
3. click on the newly created link _again_ to be able to edit it
4. click "edit link"
5. open "advanced"
6. select "open in new tab"
7. click "save"

So in anticipation of a fix, I've downloaded the latest Gutenberg 16.4 and tested the alleged solution.

While it is true that now at least the "advanced" toggle remains open when editing links after opening it once, the fact remains that it still takes considerably more clicks and time to get a link to open in a new tab than in WP 6.2.

Here's the current process after the "fix":

1. select text and "insert link" (crtl+k)
2. hit enter to create / insert the link
3. click on the newly created link _again_ to be able to edit it
4. click "edit link"
5. select "open in new tab"
6. click "save"

As you can see, the only thing that has been improved is that you don't have to click to open the "advanced" toggle each time (after you've clicked it once, that is).

But why do you still make all users go through 6 clicks, just to create a link which opens in a new tab?

Particularly, why do we have to click on the link again after creating it before even getting the "edit" option?

Why not display the check box "open in new tab" right away at step 2 as it used to be before 6.3?

So clearly, this issue has not been resolved and shouldn't be closed imho.

Furthermore, the point brought up in #[52216](https://github.com/WordPress/gutenberg/issues/52216) still remains true: 

As explained above, it does still take a lot more clicks to insert a link and get it to open in a new tab than previously.

I'm not sure why almost all threads which brought up this issue are closed, even though the issue persists and has only been improved a tiny bit with Gutenberg 16.4, but not really solved?

Thanks guys!

**Comment by draganescu** [comment]
@EW0JY thanks for the report 👏🏻 

I don't think the _number of clicks_ is something more valuable in itself than the overal experience. Reducing visual clutter is also a valuable and it's a complicated process to figure out the trade offs. The aim is to, eventually, make things seamless.

Currently with this PR tho the example you gave would be different, once you did steps 1 to 6 once, for all other links the settings drawer would remain open and the open in new tab be visible, right?

**Comment by getdave** [comment]
> this current thread here is closed

Just noting this is a PR and not an Issue. Therefore when the PR is merged the "thread" is closed. 

Myself, @richtabor and other contributors are currently exploring other ways to improve the common use case of `Opens in new tab`. For example https://github.com/WordPress/gutenberg/pull/53531 and https://github.com/WordPress/gutenberg/pull/53566

**Comment by EW0JY** [comment]
@getdave: 

> Just noting this is a PR and not an Issue. Therefore when the PR is merged the "thread" is closed.

Apologies, I'm not an experienced GitHub user, my mistake. Your explanation makes perfect sense for closing this PR.

> Myself, @richtabor and other contributors are currently exploring other ways to improve the common use case of `Opens in new tab`. For example #53531 and #53566

That's amazing!

In fact, your screen capture [here](https://github.com/WordPress/gutenberg/pull/53566#issue-1846649510) already shows the perfect solution!

As we can see in your screen grab, we would no longer have to click a second time on the link to be able to edit it and in addition to that, the "open in new tab" option is directly accessible from the preview window.

Once the reviewers have confirmed your solution, will it be available through the next Gutenberg plugin update?

Thanks so much!

**Comment by elgato91** [comment]
You are changing something that was already good enough, to make it worse. It's incredible!

Anyone working with content writing NEEDS to be able to set at least "no follow" and "open in new window" with just one click. It's absolutely insane that I've to do even just one more click to do this. These two features are very commonly used, given the impact on SEO and UX.

I don't know why you guys are doing this, but it's really nonsense. We, who work with the editor on a daily basis, simply need to be able to enter the url and check the "no follow" and "open in new window" boxes when relevant. And this must be done with lightning speed. We don't need a clean interface if it complicates our work. The number of clicks is crucial, and in doing so you are going in a useless and harmful direction. If you design user experiences, design them for people! The world is full of useless beautiful designs.

**Comment by getdave** [comment]
> Apologies, I'm not an experienced GitHub user, my mistake. Your explanation makes perfect sense for closing this PR.

Absolutely fine. Not intended as a criticism. 

> Once the reviewers have confirmed your solution, will it be available through the next Gutenberg plugin update?

If contributors are happy with the direction then the next step is to update the test coverage on the PR to ensure the tests provide coverage for the new behaviour. Once done then it will merge into the Gutenberg Plugin can land in the appropriate release. I intend to work on it today and next week but I can't promise exactly which release it will land in. 

Once merged, a `Milestone` will be assigned (see right hand sidebar on the PR) which will allow you to know which release it will land in.

> You are changing something that was already good enough, to make it worse. It's incredible!

As noted above, we've noted the feedback and are making changes to accommodate. Please bear in mind that no one is intentionally trying to make things harder but sometimes (despite our best efforts) changes have unintended consequences or impact more widely than we would have anticipated. The nature of the project means we can pivot and make changes in response to constructive feedback. In short - we are listening.

**Comment by EW0JY** [comment]
> If contributors are happy with the direction then the next step is to update the test coverage on the PR to ensure the tests provide coverage for the new behaviour. Once done then it will merge into the Gutenberg Plugin can land in the appropriate release. I intend to work on it today and next week but I can't promise exactly which release it will land in.
> 
> Once merged, a `Milestone` will be assigned (see right hand sidebar on the PR) which will allow you to know which release it will land in.

Awesome, thanks so much for the detailed explanation, Dave, I appreciate it!

Also, no rush, I understand that it needs to be tested thoroughly so if it takes a few days or a week, then that's still pretty timely, which is great.

I'll keep an eye on the milestone, like you suggested, as soon as it becomes available.

I for one am pleasantly surprised that you and the other contributors are so open to feedback and willing to make changes asap where necessary, useful and requested.

Much appreciated!

**Comment by richtabor** [comment]
> Anyone working with content writing NEEDS to be able to set at least "no follow" and "open in new window" with just one click. 

That's subjective. There's always a balance and we're consistently iterating to get closer towards that. https://github.com/WordPress/gutenberg/pull/53566 is a step in that direction, along with all the improvements outlined in #50891.

**Comment by elgato91** [comment]
As mentioned by others, I also appreciate your openness to feedback, primarily because the choices made here could potentially impact the work of millions. This isn't intended as criticism, but rather a constructive suggestion for the greater good.

> That's subjective. There's always a balance and we're consistently iterating to get closer towards that.

@richtabor 

Rich, it might be subjective, but the real question is: has anyone ever done a serious user needs analysis? Has anyone ever tried to validate the concept with a significant number of professional users?

> I don't think the number of clicks is something more valuable in itself than the overal experience. Reducing visual clutter is also a valuable and it's a complicated process to figure out the trade offs. The aim is to, eventually, make things seamless.

This, too, is deeply subjective and obviously does not stem from any scientifically validated insights. None of us likely have insights from usability testing and A/B testing. However, it's a bit counterintuitive to believe that people prefer to do things with more effort.

Personally, I seriously struggle to understand the "why" of the implementation of this feature. Basically, what could be the benefit of link preview? And which for the field for the anchor text? Does it make sense to be able to create a new blank page while I'm just inserting a link? All of these things are scattered throughout the mockups here and in #50891.

However, perhaps we should ask ourselves whether developing this really helps users, or if for them the speed and ease of completing tasks is what really matters. Perhaps along these lines, doing some serious customer discovery, maybe we might have realized that it would have been more beneficial to direct efforts towards a different aspect. In enhancing the ability to search through posts and pages when attempting to add a link, for example.

From experience, when you're developing something, you shouldn't just talk to the developers, but to the users. Otherwise, you'll end up expending a lot of energy on something that doesn't provide real benefits. The question here, in my humble opinion, isn't about releasing something that's not perfect, but about the stubbornness to persist in searching for a patch in a certain direction, instead of considering the feedback from users who clearly say: let us do things with 1 click.

**Comment by TapiwaZvaks** [comment]
@elgato91 sounds like you feel exactly how I am feeling right now. I have just spent hours researching this. Seems there is an insistence on cleaning up the UI that does not take into account ease of use. I spend maybe 8 hours per day working in WordPress and I always want all my links to open in a new tab. It's hard to imagine how that can be bad in the world of SEO. The previous setting before this latest update worked perfectly. Just as there was previously an "i" to the top left of the screen that showed the the header structure of a page, but that's hidden as well. Anyway, that's another issue altogether. 

Back to the open links in new tab issue, the entire debate here and elsewhere reminds me of a story that I heard of programmers who were hired to program earthmoving equipment (to use automation instead of humans). The programmers were very good at their work (and much appreciated), but since they had never used earthmoving equipment, they programmed in unnecessarily complex steps. They didn't understand the user environment. 

The entire project had to be scrapped and people went back to operating the graders, etc. This change feels as if it was pushed by someone that does not spend even a few minutes inside WordPress creating posts. If you did you would understand the frustration. For us, a clean interface is one that simplifies tasks. Having 6 clicks where there were previously two is very much cluttered. As it is, I may have to revert to creating posts in Microsoft Word and then pasting them to WordPress. That change has added loads of frustration to my work. Sorry if I sound a bit caustic. I had to get that off my chest, it was bugging me for a while...

**Comment by getdave** [comment]
Thanks to everyone for your feedback. As [contributors have noted above](https://github.com/WordPress/gutenberg/pull/52799#issuecomment-1674826579), we are listening and making changes in response to feedback. 

I appreciate it can be cathartic to express frustration when it appears as though decisions were made in a manner that didn't consider your use case. However, what would really help at this stage in order improve things for everyone is to review and test the PRs that are already open to address the issues noted above. Namely:

- https://github.com/WordPress/gutenberg/pull/53566
- https://github.com/WordPress/gutenberg/pull/53531

As an open source project, WordPress relies on everyone working together to improve the software. _Anyone_ is allowed to submit PRs and regular contributors will always try to be proactive in supporting with code reviews. If folks aren't able to contribute code then it's equally valuable to test PRs and check they are going in the direction that works for you.

We anticipate the Link interface will continue to undergo iterations in the near future, so it would be hugely valuable if you felt able to follow [PRs tagged with the associated label](https://github.com/WordPress/gutenberg/labels/%5BFeature%5D%20Link%20Editing) and provide feedback on their direction.

Contributors look forward to working with the wider WP community to improve things for everyone.

**Comment by elgato91** [comment]
I believe that many of us can readily understand the situation, and the fact that we are commenting here is already symptomatic that we want to help you create something good. In my case, I can contribute some feedback on UX/Interaction Design, but honestly it seems to me that the entire Link UI design is against human-centered design principles, with useless functionality detrimental to the overall user experience. Unfortunately, many of us have only come to realize this now that we've been directly impacted by the issue.

We can also comment on every single attempt to make this new UI better, but I think we should honestly acknowledge that there has been a significant regression from the old Link UI, which was better designed to be easier to use.

We're insisting on cleaning up the interface, hiding the only controls that make sense. But we are also introducing a link preview. Who needs this? What advantages does it bring? Unfortunately, there has been no response on this.

Similarly, we're implementing a text field for modifying the anchor text of a link, which is essentially a redundant function that can already be performed directly within the editor. Once again, we need to consider: Who benefits from this? What purpose does it serve?

Guys, cleaning the interface in this way is a huge misstep. To achieve a purely aesthetic result, we are adding complexity to the interface and introducing unnecessary steps. A multi-level, multi-step interface is more complex. You cannot deny this. Even the choice of introducing advanced controls in the Link UI is flawed from the outset, and increases the general complexity of the WP user experience, because it would be more intuitive to place all the advanced controls in the sidebar, akin to other blocks.

We can continue to give you feedback on your attempts to make it work, but be honest. This new Link UI is poorly designed. It doesn't take into account use cases at all, it highlights a lack of understanding of how WP is used by users and neglects basic principles of UX and ergonomics. It's the sort of outcome one might expect from novices, not from a globally collaborative project like WordPress. We need to approach this situation with honesty and humility and explore alternative paths: either reverting to the previous Link UI or re-design something more human-friendly.

**Comment by tellthemachines** [comment]
I just cherry-picked this PR to the update/bugfixes-6-3-1 branch to get it included in the next release: a3d7887627

**Comment by pb7107** [comment]
Just some feedback from a non-dev but a heavy WP user. If someone doesn't use new tab/nofollow/sponsored settings and just pastes links as is, they can do that and simply ignore the settings at the bottom. How does it help users to hide them away? I don't understand that logic. I think in this case, it's pretty clear that these link settings should be easy and quick to access, as they were before 6.3. I'm not seeing any "balance" that needs to be struck here. Just don't bury them behind several additional clicks.

**Comment by thewhitedrewcarey** [comment]
I'm 100% on board with what pb7107 mentioned above. I'm no dev, so most of what the smarter people are commenting here is well over my head. But I run a music news website and literally _every_ post we write up has at least one, if not more, external links to a band's preorder, website, link page, etc. We ALWAYS want these links to open in a new tab, and there is very little reason at all to hide these "advanced options" that were there before. 

It's not hyperbole for me to say this this is going to be huge PITA.

**Comment by TapiwaZvaks** [comment]
@thewhitedrewcarey it's exactly what we have been saying all along. As another heavy user who works on several websites, I always want my links to open in a new tab. As a web user, I also always want the links that I am clicking on to open in a new tab. I don't want to be taken away from what I am reading simply because I have clicked on a referenced article. The situation that was there before 6.3 was perfect. Unfortunately it doesn't seem like there is going back to that. Fortunately something is being done, at least for the new tab issue. But for all the other settings hidden under Advanced? Well... I repeat, the link preview that is being given prominence here does nothing. Maybe it's the one that should be hidden under Advanced.

**Comment by thewhitedrewcarey** [comment]
@TapiwaZvaks 

I'll say this much: I just updated last night, have been using it last night and today, and I absolutely hate it. I don't know who it serves to remove this functionality, because it 100% was not hurting anyone at all to have these options immediately present when copy paste a link over text. 

This is a prime example of a solution looking for a problem.

**Comment by hypnoticHODL** [comment]
Noob dev here, but long-time user of WordPress with my team of 10. I created a profile just to comment on this thread. Because everyone's been complaining about this new link feature: Project Managers, Copywriters, and Content Editors. It doesn't make sense to have to click 5 times to simply open a link in a new tab. We've been churning hundreds of posts on a monthly basis, and we feel the pain. 

The behavior in the previous WP version was way more user-friendly. And I hope this will be fixed in a coming version.

**Comment by draganescu** [comment]
Hi @beixco and thank you for participating and offering your perspective. Improving the linking flows is still work in progress and all contribution is informed by helpful replies like yours.

In [recent mock-up explorations](https://github.com/WordPress/gutenberg/issues/50891#issuecomment-1681878340) just one more click is required, but also a global prefference is suggested - so most of the times not even that click is needed.

In the mean time, the original issues that this PR here tried to solved does not exist anymore (the original change where one had to create, click edit, click advanced, click save - is gone: now the advanced section stays open once it was open and also open in new tab is temporarily surfaced to be one click away).

**Comment by hypnoticHODL** [comment]
Thanks for sharing the mock-up explorations @draganescu - and sorry if I posted in the wrong thread, I followed a link from the WordPress forum.

**Comment by draganescu** [comment]
It's perfectly fine @beixco  🙇🏻

**Pull request merged**
